### PR TITLE
Upgrade to Polkadot v1.10.0 / Refactor config

### PR DESCRIPTION
**Pull request by benoitdevos**
* Upgrade to Polkadot v1.10.0 (companion of https://github.com/logion-network/logion-pallets/pull/65)
* Split runtime code into
  * runtime construction
  * API's
  * configuration (further split into 3 parts: base, xcm and logion-specific)

logion-network/logion-internal#1283

### CODE REVIEW

File: runtime/src/configs/mod.rs [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+impl Contains<RuntimeCall> for BaseCallFilter {
+	fn contains(call: &RuntimeCall) -> bool {
+		match call {
+			RuntimeCall::Recovery(pallet_recovery::Call::create_recovery{ .. }) => false,
+			RuntimeCall::Multisig(pallet_multisig::Call::approve_as_multi{ .. }) => false,
+			RuntimeCall::Multisig(pallet_multisig::Call::as_multi{ .. }) => false,
+			_ => true
+		}
+	}
+}
```
Comment (gdethier) [inline runtime/src/configs/mod.rs:107]: Our BaseCallFilter is quite specific, to be moved to `logion_config` as well?

File: runtime/src/configs/mod.rs [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+			Balances::resolve_creating(&LogionTreasuryPalletId::get().into_account_truncating(), reward);
+		}
+	}
+
+	fn payout_to(reward: NegativeImbalance, account: &AccountId) {
+		if reward != NegativeImbalance::zero() {
+			Balances::resolve_creating(account, reward);
+		}
+	}
+}
```
Comment (gdethier) [inline runtime/src/configs/mod.rs:270]: The `RewardDistributor` and distribution keys could be moved to their own module.

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -134,6 +101,11 @@ pub type UncheckedExtrinsic =
 /// Extrinsic type that has already been checked.
 pub type CheckedExtrinsic = generic::CheckedExtrinsic<AccountId, RuntimeCall, SignedExtra>;
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo, Copy, MaxEncodedLen)]
+pub enum Region {
+	Europe,
+}
+
```
Comment (gdethier) [inline runtime/src/lib.rs:108]: Did you try to declare the type in the same way as `SS58Prefix` below?

**Review by gdethier — APPROVED** [review]
Great job! I left a couple of remarks that we can discuss.

File: runtime/src/configs/mod.rs (hunk 1 above)
Comment (benoitdevos) [reply runtime/src/configs/mod.rs:107]: Thanks for your suggestions. I'll address them in an upcoming PR.

File: runtime/src/lib.rs (hunk 3 above)
Comment (benoitdevos) [reply runtime/src/lib.rs:108]: Yes. But that did not change the generated Typescript type.

**Pull request merged**